Fix KMS encryption script to run in GCP cloud shell

Edwin-Pau opened the pull request:
Applied several fixes that enable the KMS encryption script under tools to be run from GCP cloud shell directly. A use-case would be if a user wanted to use the script to decrypt the terraform.tfvars file after completing the GCP quickstart deployment. 

Signed-off-by: Edwin-Pau <epau@teradici.com>

Commit:
Fix module import error for KMS encryption script

Added a function import_or_install_module(pip_package, module_name)
which attempts to import a specified Python package, and if it fails
it will attempt to install it first before importing it again.

This function takes in two arguments: the first one is the Python
package name and the second optional parameter is for the import name
(if the import name is different than the package name, e.g.
"google-cloud-kms" for pip3, "google.cloud.kms_v1" for import.

Signed-off-by: Edwin-Pau <epau@teradici.com>

Commit:
GCP: Use absolute paths in quickstart

Modified the GCP quickstart script to pass in absolute paths instead
of relative paths for the following terraform.tfvars variables:
gcp_credentials_file, cac_admin_ssh_pub_key_file and
centos_admin_ssh_pub_key_file.

This change allows the KMS encryption script to work in both the local
and cloud environments. This is also in-line with the terraform.tfvars
.sample file.

Signed-off-by: Edwin-Pau <epau@teradici.com>

Commit:
Update docs for absolute paths

Updated both the GCP and AWS docs under the "Customizing
terraform.tfvars" section to explicitly state that absolute paths
should be used instead of implicitly through the examples shown.

Fixed a typo under the GCP quickstart tutorial.

Signed-off-by: Edwin-Pau <epau@teradici.com>

Commit:
Fix encryption script shebang line

Fixed the shebang line for the KMS encryption script to use /usr/bin/env
which uses the user's path to run the script, making it more portable
to different environments, such as in the Google Cloud Shell.

Signed-off-by: Edwin-Pau <epau@teradici.com>

shermanyin commented on tools/kms_secrets_encryption.py:
```diff
@@ -7,13 +7,50 @@
+        print("Successfully imported {}.".format(module_name))
+
+    except ImportError:
+        install_cmd = "pip3 install {} --user".format(pip_package)
```
use f strings where you can in your commits. easier to read.

shermanyin commented on tools/kms_secrets_encryption.py:
```diff
@@ -7,13 +7,50 @@
 import os
-from google.cloud    import kms_v1
+import subprocess
 from google.oauth2   import service_account
```
no longer needed

shermanyin commented on tools/kms_secrets_encryption.py:
```diff
@@ -7,13 +7,50 @@
+    except Exception as err:
+        print("An exception occurred importing {}.".format(module_name))
+        print("{}\n".format(err))
+        raise SystemExit()
```
you can put the error string in SystemExit()

shermanyin commented on quickstart/gcp-cloudshell-quickstart.py:
```diff
@@ -378,21 +378,24 @@ def kms_encode(key, text):
 
     print('Deploying with Terraform...')
+    # Paths passed into terraform.tfvars should be absolute paths
+    cwd = os.getcwd() + '/'
```
probably better to group this with others in line 298.

shermanyin commented on tools/kms_secrets_encryption.py:
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/env python3
```
change this for gcp-cloudshell-quickstart.py too?

shermanyin reviewed (changes requested):
have a look at https://realpython.com/python-f-strings/

Commit:
Use f-strings instead of str.format()

Replaced all usage of str.format() with f-strings instead in the
KMS encryption script.

Signed-off-by: Edwin-Pau <epau@teradici.com>

Commit:
Move error string to SystemExit()

Moved error strings to SystemExit() calls instead of using print
statements.

Signed-off-by: Edwin-Pau <epau@teradici.com>

Commit:
GCP: Fix quickstart script

Fixed GCP quickstart shebang line to use /usr/bin/env
which uses the user's path to run the script, making it more portable
to different environments.

Grouped the current working directory variable (cwd) with the other
os function calls.

Signed-off-by: Edwin-Pau <epau@teradici.com>

Commit:
Remove static imports

Removed static importing of Google modules as there's a custom function
which handles the install and imports of these Google modules.

Signed-off-by: Edwin-Pau <epau@teradici.com>
